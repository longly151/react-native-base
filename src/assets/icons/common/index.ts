import loginBackground from './loginBackground.svg';

const IconCommonSource = {
  loginBackground,
};

export default IconCommonSource;
