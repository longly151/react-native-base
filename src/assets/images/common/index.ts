const ImageCommonSource = {
  placeholderImage: require('./placeholderImage.png'),
  errorImage: require('./errorImage.png'),
  loginBackground: require('./loginBackground.png'),
};

export default ImageCommonSource;
