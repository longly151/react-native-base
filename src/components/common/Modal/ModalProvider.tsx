import AppView from '@utils/AppView';
import Helper from '@utils/Helper';
import React, { useRef, useState } from 'react';
import Modal, { ModalProps, ModalType } from './index';

interface IModalContext {
  setModal: (key: Exclude<string, 'alert'>, modalProps: ModalProps) => void;
  getModal: (key: Exclude<string, 'alert'>) => React.MutableRefObject<ModalType | null>['current'];
  alert: (props: ModalProps['alert']) => void;
}
export const ModalContext = React.createContext({} as IModalContext);

const ModalProvider: React.FC<{ children: React.ReactNode }> = props => {
  const alertRef = useRef<ModalType>(null);
  const modalRefs = useRef([] as any);

  const { children } = props;

  const [alertProps, setAlertProps] = useState<ModalProps['alert']>({});
  const [modals, setModals] = useState<JSX.Element[]>([]);

  const setModal = (key: string, modalProps: ModalProps) => {
    setModals(oldModals => {
      const newModals = [...oldModals];
      let hasReplace = false;
      modals.forEach((e, i) => {
        if (e.key === key) {
          hasReplace = true;
          if (!Helper.shallowEqual(modals[i].props, modalProps)) {
            const nodeWithNewProps = React.cloneElement(modals[i], modalProps);
            newModals[i] = nodeWithNewProps;
          }
        }
      });

      if (!hasReplace) {
        const newModal = (
          <Modal ref={ref => (modalRefs.current[key] = ref)} key={key} {...modalProps} />
        );
        newModals.push(newModal);
      }
      return newModals;
    });
  };

  const getModal = (key: string) => {
    if (key === 'alert') {
      return alertRef.current;
    } else {
      return modalRefs.current[key];
    }
  };

  const alert = (newAlertProps: ModalProps['alert']) => {
    setAlertProps(newAlertProps);
    if (newAlertProps?.loading === false) {
      alertRef.current?.trigger(false);
    } else {
      alertRef.current?.trigger(true);
    }
  };

  return (
    <ModalContext.Provider
      value={{
        setModal,
        getModal,
        alert,
      }}
    >
      {children}
      {modals}
      <Modal
        testID="ModalProvider"
        ref={alertRef}
        alert={alertProps}
        statusBarTranslucent
        deviceHeight={
          AppView.windowHeight + AppView.safeAreaInsets.top + AppView.safeAreaInsets.bottom
        }
      />
    </ModalContext.Provider>
  );
};

export default ModalProvider;

export const ModalConsumer = ModalContext.Consumer;
