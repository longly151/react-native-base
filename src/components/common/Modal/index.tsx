import { StyleProp, View, ViewStyle } from 'react-native';
import React, { useEffect, useImperativeHandle, useRef, useState } from 'react';
import withTailwind from '@components/hoc/withTailwind';
import RNModal, { ModalProps as RNModalProps } from 'react-native-modal';
import { useTheme } from '@react-navigation/native';
import { useTranslation } from 'react-i18next';
import Text from '../Text';
import AppView from '@utils/AppView';
import Button from '../Button';
import { Icon, Loading } from '@components';

export interface ModalProps extends Partial<RNModalProps> {
  alert?: {
    title?: string;
    description?: string;
    type?: 'success' | 'error';
    loading?: boolean;
    onOkButtonPress?: Function;
    onCancelButtonPress?: Function;
    okTitle?: string;
    cancelTitle?: string;
    reverseButtonPosition?: boolean;
    showCancel?: boolean;
    width?: number;
    height?: number;
    buttonStyle?: StyleProp<ViewStyle>;
  };
  className?: string;
  backdropClose?: boolean;
}

const Modal: React.FC<Omit<ModalProps, 'children'> & { children: React.ReactNode }> =
  React.forwardRef((props, ref: any) => {
    const modalRef = useRef<RNModal>(null);

    const theme = useTheme();
    const { t } = useTranslation();

    const {
      style: styleProp,
      isVisible: isVisibleProp,
      onBackdropPress: onBackdropPressProp,
      backdropClose: backdropCloseProp,
      alert,
      ...otherProps
    } = props;

    const backdropClose = alert?.loading ? false : backdropCloseProp;

    const style: StyleProp<ViewStyle> = [styleProp];

    const [isVisible, setIsVisible] = useState(isVisibleProp);
    const [forwardData, setForwardData] = useState(null);

    const trigger = (nextIsVisible?: boolean) => {
      if (nextIsVisible === undefined) {
        setIsVisible(!isVisible);
      } else {
        setIsVisible(nextIsVisible);
      }
    };

    const renderChildren = () => {
      const listOfProps: any = {
        close: () => setIsVisible(false),
      };
      if (forwardData) {
        listOfProps.forwardData = forwardData;
      }
      const nodeWithProps = React.cloneElement(otherProps.children as any, listOfProps);
      return nodeWithProps;
    };

    useEffect(() => {
      setIsVisible(isVisibleProp);
    }, [isVisibleProp]);

    useImperativeHandle(ref, () => ({
      ...modalRef.current,
      trigger,
      setForwardData,
    }));

    if (alert) {
      otherProps.animationIn = 'zoomIn';
      otherProps.animationOut = 'zoomOut';
      otherProps.animationInTiming = 150;
      otherProps.animationOutTiming = 150;
      otherProps.backdropTransitionInTiming = 150;
      otherProps.backdropTransitionOutTiming = 0;

      const modalWidth = alert.width || AppView.bodyWidth;
      const onCancel = () => {
        setIsVisible(false);
        if (alert.onCancelButtonPress) {
          alert.onCancelButtonPress();
        }
      };

      const onOk = () => {
        setIsVisible(false);
        if (alert.onOkButtonPress) {
          alert.onOkButtonPress();
        }
      };

      if (typeof alert.loading !== 'undefined') {
        otherProps.children = (
          <View className="bg-color-card py-5 px-7 rounded self-center justify-center">
            <Loading testID="Loading" size="large" className="mt-1" />
            <Text className="text-color-primary text-center text-lg font-bold my-1">
              {alert.title || t('common.loading')}
            </Text>
          </View>
        );
      } else {
        otherProps.children = (
          <View
            className="bg-color-card p-4 rounded self-center justify-center"
            style={{
              width: modalWidth,
              height: alert.height,
            }}
          >
            {alert.type === 'success' || alert.type === 'error' ? (
              <Icon
                testID={alert.type === 'success' ? 'SuccessIcon' : 'ErrorIcon'}
                type="antdesign"
                name={alert.type === 'success' ? 'checkcircle' : 'closecircle'}
                className="self-center my-2"
                color={alert.type === 'success' ? theme.colors.success : theme.colors.error}
                style={{ transform: [{ rotateZ: '-5deg' }] }}
                size={50}
              />
            ) : null}
            {alert.title ? (
              <Text
                testID="TitleText"
                className="text-center text-xl font-semibold my-1"
                color={!alert.type ? theme.colors.primary : theme.colors.grey0}
              >
                {alert.title}
              </Text>
            ) : null}
            {alert.description ? (
              <Text
                testID="DescriptionText"
                className="text-center text-base my-1"
                color={theme.colors.grey0}
              >
                {alert.description}
              </Text>
            ) : null}
            <View
              style={{
                flexDirection: alert.reverseButtonPosition ? 'row-reverse' : 'row',
                justifyContent: !alert.showCancel ? 'center' : 'space-between',
                alignItems: 'center',
                paddingVertical: 16,
              }}
            >
              {alert.showCancel ? (
                <Button
                  testID="CancelButton"
                  title={alert.cancelTitle || t('common.cancel')}
                  outline={!alert.reverseButtonPosition}
                  style={[
                    {
                      width: (modalWidth - 50) / 2,
                      alignItems: 'center',
                    },
                    alert.buttonStyle,
                  ]}
                  onPress={onCancel}
                />
              ) : null}
              <Button
                testID="OkButton"
                title={alert.okTitle || t('common.ok')}
                outline={alert.reverseButtonPosition}
                style={[
                  {
                    width: !alert.showCancel ? modalWidth - 50 : (modalWidth - 50) / 2,
                    alignItems: 'center',
                  },
                  alert.buttonStyle,
                ]}
                onPress={onOk}
              />
            </View>
          </View>
        );
      }
    }

    const onBackdropPress = () => {
      if (backdropClose) {
        setIsVisible(false);
      }
      onBackdropPressProp?.();
    };

    return (
      <RNModal
        testID="Modal"
        ref={modalRef}
        statusBarTranslucent={!!alert}
        deviceHeight={
          alert
            ? AppView.windowHeight + AppView.safeAreaInsets.top + AppView.safeAreaInsets.bottom
            : undefined
        }
        backdropTransitionOutTiming={0}
        {...otherProps}
        isVisible={isVisible}
        onBackdropPress={onBackdropPress}
        style={style}
      >
        {renderChildren()}
      </RNModal>
    );
  });

Modal.defaultProps = {
  backdropClose: true,
};

export interface ModalType extends RNModal {
  trigger: (nextIsVisible?: boolean) => void;
  setForwardData: (newForwardData: any) => void;
}

export default withTailwind(Modal) as React.FC<ModalProps & { ref?: React.LegacyRef<RNModal> }>;
