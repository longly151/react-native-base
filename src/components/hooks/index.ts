export { default as useRequest } from './useRequest';
export { default as useListRequest } from './useListRequest';
export { default as useMutation } from './useMutation';
export { default as useModal } from './useModal';
export { default as useBottomSheet } from './useBottomSheet';
export { default as usePermission } from './usePermission';
