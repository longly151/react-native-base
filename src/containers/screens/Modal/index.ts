import DefaultModal from './DefaultModal';

const ModalScreens = {
  DefaultModal,
};

export default ModalScreens;
