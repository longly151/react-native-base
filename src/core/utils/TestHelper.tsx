import React, { useLayoutEffect } from 'react';
import { render, RenderAPI } from '@testing-library/react-native';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import store, { persistor, createStore } from '@configs/store';
import { LoadingScreen } from '@app';
import { NavigationContainer } from '@react-navigation/native';
import Navigation from './Navigation';
import MockAdapter from 'axios-mock-adapter';
import { Axios } from '@utils/Api';

import DefaultColor from '@themes/DefaultColor';
import { RootState, useAppDispatch, useAppSelector } from './Redux';
import _ from 'lodash';

import { renderHook } from '@testing-library/react-hooks';

type DeepPartial<T> = T extends object
  ? {
      [P in keyof T]?: DeepPartial<T[P]>;
    }
  : T;

/**
 * Login Handler
 */
export const successCredential = { email: 'admin@gmail.com', password: '123456' };
export const successLoginResponse = {
  data: {
    token:
      'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpZCI6MSwic3RhdHVzIjoiQUNUSVZFIiwicm9sZSI6IkFETUlOIiwicm9sZUlkIjoxLCJwZXJtaXNzaW9ucyI6WyJBTEwiXSwidGFibGUiOiJ1c2VycyIsImlkRm9yZWlnbktleSI6InVzZXJJZCIsImlhdCI6MTY1NTQ1NzU5OCwiZXhwIjoxNjYwNjQxNTk4fQ.ZA1dsxArHAjb60HzQgKbkinl49cf5OiDWASiE_Wr0SA',
    id: 1,
    email: 'admin@gmail.com',
    fullName: 'Admin',
    phone: '0327571918',
    avatar:
      'https://encrypted-tbn0.gstatic.com/images?q=tbn%3AANd9GcRlUbAyS_643dq_B69jZAlPNW6_Xc7SLELY6SpRsc5OI2wHiiYG&usqp=CAU',
    role: {
      createdAt: '2021-04-19T05:58:23.094Z',
      updatedAt: '2021-04-19T05:58:23.094Z',
      deletedAt: null,
      id: 1,
      name: 'ADMIN',
      description: 'ADMIN',
      permissions: [
        {
          id: 1,
          name: 'ALL',
          description: null,
        },
      ],
    },
  },
};

export const failCredential = { email: 'client@gmail.com', password: '123456' };
export const failLoginResponse = {
  statusCode: 401,
  message: [
    {
      code: 'INCORRECT_EMAIL_PASSWORD',
      description: 'Incorrect email or password',
    },
  ],
  error: 'Unauthorized',
};

const auth = {
  loading: false,
  data: successLoginResponse.data,
  error: null,
};
const withAuthStore = createStore({ auth });

/**
 * Wrapper handler
 */

export type OtherRenderOptions = {
  preloadedState?: DeepPartial<RootState>;
  withAuth?: boolean;
  disableNavigationRef?: boolean;
};

const AppContainer = ({
  children,
  disableNavigationRef,
}: {
  children: any;
  disableNavigationRef?: boolean;
}) => {
  const [loginData] = useAppSelector(state => [state.auth.data]);

  const dispatch = useAppDispatch();

  useLayoutEffect(() => {
    const initApp = () => {
      if (loginData?.token) {
        global.token = loginData.token;
      }
      global.fn = {} as any;
      global.dispatch = dispatch;
    };
    initApp();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);
  return (
    <NavigationContainer
      ref={disableNavigationRef ? null : Navigation.navigationRef}
      theme={DefaultColor}
    >
      {children}
    </NavigationContainer>
  );
};

const AppProviders = ({
  children,
  preloadedState,
  withAuth,
  disableNavigationRef,
}: {
  children: any;
} & OtherRenderOptions) => {
  const mergedPreloadedState: DeepPartial<RootState> = withAuth
    ? _.merge({ auth }, preloadedState as any)
    : preloadedState;

  return (
    // Create new store if preloadedState !== undefined
    <Provider
      store={preloadedState ? createStore(mergedPreloadedState) : withAuth ? withAuthStore : store}
    >
      <PersistGate loading={LoadingScreen()} persistor={persistor}>
        <AppContainer disableNavigationRef={disableNavigationRef}>{children}</AppContainer>
      </PersistGate>
    </Provider>
  );
};

const customRender = (
  component: React.ReactElement<any>,
  // options?: RenderOptions & OtherRenderOptions,
  options?: OtherRenderOptions,
): RenderAPI => {
  const CustomWrapper = options?.wrapper || React.Fragment;
  const Wrapper = options?.wrapper
    ? ({ children }: any) => (
        <AppProviders
          preloadedState={options?.preloadedState}
          withAuth={options?.withAuth}
          disableNavigationRef={options.disableNavigationRef}
        >
          <CustomWrapper>{children}</CustomWrapper>
        </AppProviders>
      )
    : ({ children }: any) => (
        <AppProviders
          preloadedState={options?.preloadedState}
          withAuth={options?.withAuth}
          disableNavigationRef={options?.disableNavigationRef}
        >
          {children}
        </AppProviders>
      );

  return render(component, { ...options, wrapper: Wrapper });
};

const customRenderHook: typeof renderHook = (callback, options) => {
  const disableNavigationRef =
    typeof options?.disableNavigationRef === 'undefined' ? true : options?.disableNavigationRef;
  const CustomWrapper: any = options?.wrapper || React.Fragment;
  const Wrapper = options?.wrapper
    ? ({ children }: any) => (
        <AppProviders
          preloadedState={options?.preloadedState}
          withAuth={options?.withAuth}
          disableNavigationRef={disableNavigationRef}
        >
          <CustomWrapper>{children}</CustomWrapper>
        </AppProviders>
      )
    : ({ children }: any) => (
        <AppProviders
          preloadedState={options?.preloadedState}
          withAuth={options?.withAuth}
          disableNavigationRef={disableNavigationRef}
        >
          {children}
        </AppProviders>
      );

  return renderHook(callback, { ...options, wrapper: Wrapper });
};

// re-export everything
export * from '@testing-library/react-native';

// override render method
export { customRender as render };

export { customRenderHook as renderHook };

const mockAxios = new MockAdapter(Axios, { delayResponse: 200 });

export { mockAxios };
