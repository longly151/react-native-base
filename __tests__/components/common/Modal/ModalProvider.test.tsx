import React from 'react';
import { fireEvent, render, act, renderHook } from '@utils/TestHelper';
import { useModal } from '@components/hooks';
import { Text, View } from 'react-native';
import ModalProvider from '@components/common/Modal/ModalProvider';
import { ModalType } from '@components/common/Modal';

test('should render correctly', () => {
  const { toJSON, getByTestId } = render(<React.Fragment />, { wrapper: ModalProvider });
  expect(toJSON()).toMatchSnapshot();
  expect(getByTestId('ModalProvider')).toBeTruthy();
});

test('should show alert modal correctly', async () => {
  const { result } = renderHook(() => useModal(), { wrapper: ModalProvider });

  const onOkButtonPress = jest.fn();
  const onCancelButtonPress = jest.fn();

  /**
   * Alert Modal
   */
  act(() => {
    result.current.alert({
      title: 'Confirmation',
      description: 'Are you sure you want to log out?',
      okTitle: 'Logout',
      showCancel: true,
      reverseButtonPosition: true,
      onOkButtonPress,
      onCancelButtonPress,
    });
  });

  let alertModalProps = result.current.getModal('alert')?.props;

  expect(alertModalProps?.isVisible).toBe(true);

  const { toJSON, getByTestId, rerender } = render(alertModalProps?.children as any);
  expect(toJSON()).toMatchSnapshot();

  expect(getByTestId('TitleText')).toHaveTextContent('Confirmation');
  expect(getByTestId('DescriptionText')).toHaveTextContent('Are you sure you want to log out?');
  expect(getByTestId('OkButton')).toHaveTextContent('Logout');

  fireEvent.press(getByTestId('CancelButton'));
  expect(onCancelButtonPress).toBeCalledTimes(1);

  fireEvent.press(getByTestId('OkButton'));
  expect(onOkButtonPress).toBeCalledTimes(1);

  onOkButtonPress.mockClear();
  onCancelButtonPress.mockClear();

  /**
   * Loading Modal
   */
  act(() => {
    result.current.alert({
      loading: true,
    });
  });
  alertModalProps = result.current.getModal('alert')?.props;

  rerender(alertModalProps?.children as any);
  expect(toJSON()).toMatchSnapshot();

  expect(alertModalProps?.isVisible).toBe(true);
  expect(getByTestId('Loading')).toBeTruthy();

  act(() => {
    result.current.alert({
      loading: false,
    });
  });
  alertModalProps = result.current.getModal('alert')?.props;
  expect(alertModalProps?.isVisible).toBe(false);

  /**
   * Success/Error Modal
   */
  act(() => {
    result.current.alert({
      title: 'Submission successful',
      type: 'success',
      onOkButtonPress,
    });
  });

  alertModalProps = result.current.getModal('alert')?.props;
  expect(alertModalProps?.isVisible).toBe(true);

  rerender(alertModalProps?.children as any);
  expect(toJSON()).toMatchSnapshot();

  expect(getByTestId('TitleText')).toHaveTextContent('Submission successful');
  expect(getByTestId('SuccessIcon')).toBeTruthy();

  fireEvent.press(getByTestId('OkButton'));
  expect(onOkButtonPress).toBeCalledTimes(1);

  alertModalProps = result.current.getModal('alert')?.props;
  expect(alertModalProps?.isVisible).toBe(false);

  act(() => {
    result.current.alert({
      title: 'Submission failed',
      type: 'error',
      onOkButtonPress,
    });
  });

  alertModalProps = result.current.getModal('alert')?.props;
  rerender(alertModalProps?.children as any);
  expect(toJSON()).toMatchSnapshot();

  expect(getByTestId('TitleText')).toHaveTextContent('Submission failed');
  expect(getByTestId('ErrorIcon')).toBeTruthy();
});

test('should show modal correctly', () => {
  /**
   * Init New Modal
   */
  const { result } = renderHook(() => useModal(), { wrapper: ModalProvider });

  const MODAL_NAME = 'DefaultModal';
  act(() => {
    result.current.setModal(MODAL_NAME, {
      children: <View testID="ModalContent" />,
    });
  });

  let modal = result.current.getModal(MODAL_NAME) as ModalType;

  const { toJSON, getByTestId, rerender } = render(modal.props?.children as any);

  expect(toJSON()).toMatchSnapshot();

  expect(getByTestId('ModalContent')).toBeTruthy();

  // Open Modal
  act(() => {
    modal.trigger();
  });
  modal = result.current.getModal(MODAL_NAME) as ModalType;
  expect(modal.props?.isVisible).toBe(true);

  // Close Modal
  act(() => {
    modal.trigger();
  });
  modal = result.current.getModal(MODAL_NAME) as ModalType;
  expect(modal.props?.isVisible).toBe(false);

  /**
   * Update Modal
   */
  act(() => {
    result.current.setModal(MODAL_NAME, {
      children: <Text testID="NewModalContent" />,
    });
  });

  modal = result.current.getModal(MODAL_NAME) as ModalType;
  rerender(modal.props?.children as any);
  expect(getByTestId('NewModalContent')).toBeTruthy();
});
