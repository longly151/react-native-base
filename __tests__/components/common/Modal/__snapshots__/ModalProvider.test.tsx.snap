// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<Modal
  animationType="none"
  deviceHeight={null}
  deviceWidth={null}
  hardwareAccelerated={false}
  hideModalContentWhileAnimating={false}
  onBackdropPress={[Function]}
  onModalHide={[Function]}
  onModalWillHide={[Function]}
  onModalWillShow={[Function]}
  onRequestClose={[Function]}
  panResponderThreshold={4}
  scrollHorizontal={false}
  scrollOffset={0}
  scrollOffsetMax={0}
  scrollTo={null}
  statusBarTranslucent={true}
  supportedOrientations={
    Array [
      "portrait",
      "landscape",
    ]
  }
  swipeThreshold={100}
  testID="ModalProvider"
  transparent={true}
  visible={false}
/>
`;

exports[`should show alert modal correctly 1`] = `
<View
  close={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "height": undefined,
      "justifyContent": "center",
      "padding": 16,
      "width": 710,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#219EE5",
          "fontFamily": "GoogleSans-Regular",
        },
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "lineHeight": 28,
            "marginBottom": 4,
            "marginTop": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
    testID="TitleText"
  >
    Confirmation
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#393e42",
          "fontFamily": "GoogleSans-Regular",
        },
        Array [
          Object {
            "fontSize": 16,
            "lineHeight": 24,
            "marginBottom": 4,
            "marginTop": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
    testID="DescriptionText"
  >
    Are you sure you want to log out?
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row-reverse",
        "justifyContent": "space-between",
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {},
            Object {
              "elevation": 4,
              "flexGrow": 1,
              "height": undefined,
              "justifyContent": "center",
              "overflow": "visible",
              "width": 330,
            },
          ],
        ]
      }
      testID="ContainerView"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": undefined,
            "opacity": 1,
          }
        }
        testID="CancelButton"
      >
        <View
          style={
            Array [
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#219EE5",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": 330,
                },
              ],
            ]
          }
          testID="TouchableView"
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "GoogleSans-Regular",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "margin": 12,
                  "opacity": 1,
                  "textAlign": "center",
                },
                Object {},
                undefined,
              ]
            }
            testID="ButtonTitleText"
          >
            Cancel
          </Text>
        </View>
      </View>
    </View>
    <View
      style={
        Array [
          Array [
            Object {},
            Object {
              "elevation": 0,
              "flexGrow": 1,
              "height": undefined,
              "justifyContent": "center",
              "overflow": "visible",
              "width": 331,
            },
          ],
        ]
      }
      testID="ContainerView"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": undefined,
            "opacity": 1,
          }
        }
        testID="OkButton"
      >
        <View
          style={
            Array [
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "transparent",
                  "borderColor": "#219EE5",
                  "borderRadius": 8,
                  "borderWidth": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": 330,
                },
              ],
            ]
          }
          testID="TouchableView"
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "GoogleSans-Regular",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "margin": 12,
                  "opacity": 1,
                  "textAlign": "center",
                },
                Object {
                  "color": "#219EE5",
                },
                undefined,
              ]
            }
            testID="ButtonTitleText"
          >
            Logout
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should show alert modal correctly 2`] = `
<View
  close={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "justifyContent": "center",
      "paddingHorizontal": 28,
      "paddingVertical": 20,
    }
  }
>
  <ActivityIndicator
    color="#219EE5"
    size="large"
    style={
      Array [
        Array [
          Object {
            "marginTop": 4,
          },
          undefined,
        ],
        Object {
          "display": "flex",
        },
      ]
    }
    testID="Loading"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#219EE5",
          "fontFamily": "GoogleSans-Regular",
        },
        Array [
          Object {
            "fontSize": 18,
            "fontWeight": "700",
            "lineHeight": 28,
            "marginBottom": 4,
            "marginTop": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
    testID="Text"
  >
    Loading
  </Text>
</View>
`;

exports[`should show alert modal correctly 3`] = `
<View
  close={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "height": undefined,
      "justifyContent": "center",
      "padding": 16,
      "width": 710,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#34BFA3",
          "fontSize": 50,
        },
        Array [
          Array [
            Object {
              "alignSelf": "center",
              "marginBottom": 8,
              "marginTop": 8,
            },
            Object {
              "transform": Array [
                Object {
                  "rotateZ": "-5deg",
                },
              ],
            },
          ],
        ],
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="SuccessIcon"
  >
    î˜°
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#393e42",
          "fontFamily": "GoogleSans-Regular",
        },
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "lineHeight": 28,
            "marginBottom": 4,
            "marginTop": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
    testID="TitleText"
  >
    Submission successful
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {},
            Object {
              "elevation": 4,
              "flexGrow": 0,
              "height": undefined,
              "justifyContent": "center",
              "overflow": "visible",
              "width": 660,
            },
          ],
        ]
      }
      testID="ContainerView"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": undefined,
            "opacity": 1,
          }
        }
        testID="OkButton"
      >
        <View
          style={
            Array [
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#219EE5",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": 660,
                },
              ],
            ]
          }
          testID="TouchableView"
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "GoogleSans-Regular",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "margin": 12,
                  "opacity": 1,
                  "textAlign": "center",
                },
                Object {},
                undefined,
              ]
            }
            testID="ButtonTitleText"
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should show alert modal correctly 4`] = `
<View
  close={[Function]}
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": "#ffffff",
      "borderRadius": 8,
      "height": undefined,
      "justifyContent": "center",
      "padding": 16,
      "width": 710,
    }
  }
>
  <Text
    allowFontScaling={false}
    selectable={false}
    style={
      Array [
        Object {
          "color": "#F4516C",
          "fontSize": 50,
        },
        Array [
          Array [
            Object {
              "alignSelf": "center",
              "marginBottom": 8,
              "marginTop": 8,
            },
            Object {
              "transform": Array [
                Object {
                  "rotateZ": "-5deg",
                },
              ],
            },
          ],
        ],
        Object {
          "fontFamily": "anticon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
    testID="ErrorIcon"
  >
    î˜®
  </Text>
  <Text
    style={
      Array [
        Object {
          "color": "#393e42",
          "fontFamily": "GoogleSans-Regular",
        },
        Array [
          Object {
            "fontSize": 20,
            "fontWeight": "600",
            "lineHeight": 28,
            "marginBottom": 4,
            "marginTop": 4,
            "textAlign": "center",
          },
          undefined,
        ],
      ]
    }
    testID="TitleText"
  >
    Submission failed
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "paddingVertical": 16,
      }
    }
  >
    <View
      style={
        Array [
          Array [
            Object {},
            Object {
              "elevation": 4,
              "flexGrow": 0,
              "height": undefined,
              "justifyContent": "center",
              "overflow": "visible",
              "width": 660,
            },
          ],
        ]
      }
      testID="ContainerView"
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "flexGrow": undefined,
            "opacity": 1,
          }
        }
        testID="OkButton"
      >
        <View
          style={
            Array [
              Array [
                Object {},
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#219EE5",
                  "borderRadius": 8,
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "width": 660,
                },
              ],
            ]
          }
          testID="TouchableView"
        >
          <Text
            style={
              Array [
                Object {
                  "alignSelf": "center",
                  "color": "white",
                  "fontFamily": "GoogleSans-Regular",
                  "fontSize": 18,
                  "fontWeight": "500",
                  "margin": 12,
                  "opacity": 1,
                  "textAlign": "center",
                },
                Object {},
                undefined,
              ]
            }
            testID="ButtonTitleText"
          >
            OK
          </Text>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`should show modal correctly 1`] = `
<View
  close={[Function]}
  testID="ModalContent"
/>
`;
